contracts/interstake-yield-generator/src/state.rs

```diff
@@ -47,14 +49,41 @@ impl StakeDetails {
         Ok(())
     }
 }
-
 #[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub struct ClaimDetails {
-    pub release_timestamp: Timestamp,
+    pub release_timestamp: Option<Timestamp>,
     pub amount: Coin,
 }
+#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct UnbondInfo {
+    pub latest: Timestamp,
+}
+
+pub const FOUR_DAYS: u64 = 4 * 60 * 60 * 24;
+
+impl UnbondInfo {
+    pub fn unbond_now(self, now: Timestamp) -> Result<UnbondInfo, ContractError> {
+        if self.latest.plus_seconds(FOUR_DAYS) <= now {
+            Ok(UnbondInfo { latest: now })
+        } else {
+            Err(ContractError::UnbondingTooSoon {})
+        }
+    }
+
+    pub fn now(now: Timestamp) -> Self {
+        UnbondInfo { latest: now }
+    }
+
+    pub fn new(now: Timestamp) -> Self {
+        UnbondInfo {
+            latest: now.minus_seconds(FOUR_DAYS),
+        }
+    }
```

Comment:
I don't like this design where user might think he creates unbond info with given timestamp, and underneath quietly that data is modified.
I don't have better idea right now, but maybe at least indicate it in the name? Something like `new_minus_four_days`, or rename `UnbondInfo` structure to reflect that?

Reply:
Absolutely... I redesigned the unbond info completely. Its gone entirely and replaced by `Expiration`